user/src/com/google/gwt/user/client/impl/WindowImpl.java

```diff
@@ -43,7 +62,9 @@ public native void initWindowCloseHandler() /*-{
       try {
         ret = $entry(@com.google.gwt.user.client.Window::onClosing())();
       } finally {
-        oldRet = oldOnBeforeUnload && oldOnBeforeUnload(evt);
+        try {
+          oldRet = oldOnBeforeUnload && oldOnBeforeUnload(evt);
+        } catch (ignored) {}
```

Comment:
Should we be brave enough and log it? In previous code the exception escaped here and might end up on developer console. Now it would definitely be swallowed.

Reply:
That's a fair point - I had intended to ensure that if `ret` was set we wouldn't cancel it, but I'm changing too much here, and should preserve those semantics.

Reply:
I like the fact that `ret` now goes through the code path no matter what. Just log the ignored error and it should be fine.

Reply:
Sorry, how do you want that to be logged? I wouldn't want to directly use console.log, and uncaught ex handler requires wrapping in Throwable... I'm going with reporting to window.onerror (which will then typically be handled by the UCEH)? Reverting seems like a better option, but let me know what you think.